src/Microsoft.DotNet.VersionTools/Dependencies/FileRegexUpdater.cs

```diff
@@ -0,0 +1,67 @@
+﻿// Licensed to the .NET Foundation under one or more agreements.
+// The .NET Foundation licenses this file to you under the MIT license.
+// See the LICENSE file in the project root for more information.
+
+using System;
+using System.Collections.Generic;
+using System.Diagnostics;
+using System.IO;
+using System.Linq;
+using System.Text;
+using System.Text.RegularExpressions;
+
+namespace Microsoft.DotNet.VersionTools.Dependencies
+{
+    public abstract class FileRegexUpdater : IDependencyUpdater
+    {
+        public string Path { get; set; }
+        public Regex Regex { get; set; }
```

Comment:
This also replaces elements in xml files, like the dependency validation properties in CoreFX/WCF. `project.json` upgrades are done in `ProjectJsonUpdater`, which is Newtonsoft.Json, so still a problem.